src/spikeinterface/core/generate.py

```diff
@@ -1690,13 +1688,13 @@ class InjectTemplatesRecording(BaseRecording):
         Can be None (no scaling).
         Can be scalar all spikes have the same factor (certainly useless).
         Can be a vector with same shape of spike_vector of the sorting.
-    parent_recording: BaseRecording | None
+    parent_recording: BaseRecording | None, default: None
         The recording over which to add the templates.
         If None, will default to traces containing all 0.
-    num_samples: list[int] | int | None
+    num_samples: list[int] | int | None, default: None
```

Comment:
```suggestion
    num_samples : list[int] | int | None, default: None
```

Reply:
Should it not be optional in the case of None?

Reply:
As in the `default: None` part? I'm not sure on this @zm711 may know, happy to make any changes as required

Reply:
Yeah, this is an issue where we had previously agreed that None would be `default: None` so much of the code base has been transitioning over to that style. But more recently we had a discussion about using `, num_samples : list[int] | int | None, optional` as an alternative style. Maybe we should add this to a maintenance meeting that way we can all discuss this in person and make sure we are all on the same page with the style. We could do a quick vote to make sure everyone agrees on which of the two styles.

Reply:
Nice this sounds good, I think I prefer `default: None` for consistency and it is clearer than `optional` if you don't know much about typing. If most of the code is `default: None` shall I leave this for now and we can discuss at the meeting?

Reply:
I'm fine with that for now :)

Reply:
All right if you find those semantics clearer.